scripts/create-kubestellar-demo-env.sh

```diff
@@ -134,14 +97,85 @@ checking_cluster() {
             
         else
             echo "$1 not found. Trying again..."
-            sleep 5
+            sleep 20
         fi
 
     done
 }
+##########################################
+
+echo -e "\nStarting environment clean up..."
+echo -e "Starting cluster clean up..."
+
+yq -i 'del(.preferences)' ${KUBECONFIG:-$HOME/.kube/config} 
+
+cluster_clean_up "kind delete cluster --name kubeflex" &
+cluster_clean_up "kind delete cluster --name cluster1" &
+cluster_clean_up "kind delete cluster --name cluster2" &
+wait
+echo -e "Cluster space clean up has been completed"
+
+echo -e "\nStarting context clean up..."
+context_clean_up
+echo "Context space clean up completed"
+
+echo -e "\nStarting the process to install KubeStellar core: kind-kubeflex..."
+clusters=(cluster1 cluster2)
+for cluster in "${clusters[@]}"; do
+   (
+     echo -e "Creating cluster ${cluster}..."
+     kind create cluster --name "${cluster}" >/dev/null 2>&1 &&
+     kubectl config rename-context "kind-${cluster}" "${cluster}" >/dev/null 2>&1
+     echo -e "${cluster} creation and context setup complete"
+   ) &
+done
+wait 
+
+echo -e "Creating KubeFlex cluster with SSL Passthrough"
+curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${kubestellar_version}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --port 9443 
+echo -e "Completed KubeFlex cluster with SSL Passthrough"
+
+echo -e "\nPulling container images local..."
+images=("ghcr.io/loft-sh/vcluster:0.16.4"
+        "rancher/k3s:v1.27.2-k3s1"
+        "quay.io/open-cluster-management/registration-operator:v0.13.2"
+        "docker.io/bitnami/postgresql:16.0.0-debian-11-r13")
+
+for image in "${images[@]}"; do
+    (
+        docker pull "$image" && kind load docker-image "$image" --name kubeflex
+    ) &
```

Comment:
FYI: Because _all_ of these are forked, there is no wait for any of them to complete before starting the `helm upgrade`. Is that what you intend?

Here is what I see in my typescript from around that time:

```
...
Completed KubeFlex cluster with SSL Passthrough

Pulling container images local...
Release "ks-core" does not exist. Installing it now.
Pulled: ghcr.io/kubestellar/kubestellar/core-chart:0.25.0-rc.1
Digest: sha256:7b50ef982b212a94a77019fc6e72ee56bd55c960490b27a813ab00c60dcb1f67
16.0.0-debian-11-r13: Pulling from bitnami/postgresql
v1.27.2-k3s1: Pulling from rancher/k3s
Digest: sha256:3331ad89ba2d1af68e36521724440638be3834978ac8288c49e54929357143e6
Status: Image is up to date for bitnami/postgresql:16.0.0-debian-11-r13
...
```

Reply:
it actually does not seem to matter, the creation of the core chart starts and takes much longer to get to the containers that are being fetched. On top of this, the pre-fetch that takes the longest is OCM. I put the commands here because I knew it was the first opportunity to fetch and then upload them to kind-kubeflex, and by the time they were done uploading to kind from docker, they would be plenty of time before they were needed. There are more that could be pre-fetched, but those are cared for by the helm chart which is why it takes a long time to install.